spec/models/course_spec.rb

```diff
@@ -37,6 +37,62 @@
       end
     end
 
+    describe 'levels' do
+      let!(:user) { create(:user, role: :administrator) }
+      let!(:course) { create(:course) }
+      let!(:levels) do
+        create_list(:course_level, 5, course: course).map(&:experience_points_threshold)
+      end
+
+      describe '.levels' do
+        it 'returns levels is ascending order' do
+          course_level_numbers = course.levels.map(&:experience_points_threshold)
+          expect(course_level_numbers).to eq levels
+        end
+      end
+
+      describe '#compute_level' do
+        it 'returns nil when experience_points is 0' do
+          level = course.compute_level(0)
+          expect(level).to be_nil
+        end
+      end
+
+      describe '#compute_level_number' do
+        context 'when experience_points is 0' do
+          it 'returns 0' do
+            level = course.compute_level_number(0)
+            expect(level).to eq 0
+          end
+        end
+
+        context 'given arbitrary experience_points value' do
+          it 'returns the correct level number' do
+            experience_points = rand(levels.max)
```

Comment:
This is for the case where the level lands in the middle, i.e. not the first nor last level. Tiny (negligible?) chance that it will turn out to be the cases where we exp coincides exactly with a level threshold.